### ISSUE TITLE
Logger's product not cleared after geom processing

### ISSUE THREAD

**Issue by Stinkfist0** [IfcOpenShell/IfcOpenShell#106]
Possible log prints after geom processing have the last processed GUID as a prefix (e.g. the model offset print).

**Comment by aothms** [comment]
@iras Issue a `SetProduct(boost::none)` somewhere? Maybe here: https://github.com/IfcOpenShell/IfcOpenShell/blob/master/src/ifcgeom/IfcGeomIterator.h#L550

**Comment by iras** [comment]
Sure. See #107.

**Comment by Stinkfist0** [comment]
BTW just an idea for the future. Instead of the current way
```c++
Logger::SetProduct(product);
Logger::Message(...);
// ...
Logger::SetProduct(0); // we must be very careful that this is called
```
it might be easier instead to do
```c++
{
    ProductLogger logger(product);
    logger.Message(...);
    // ...
}
```

**Comment by aothms** [comment]
Yeah, but the idea is that messages issued deep down in the geom interpretation also get this product prefixed to the output. So it would need to be something like this still to globally affect the behaviour of the logger. Perhaps we should look for a more general solution for this anyway, find ways to include line number and files, find ways to trace and time functions in super verbose mode. That could be very insightful potentially.

~~~
class scoped_logger_context {
    scoped_logger_context(p) {
        Logger::SetProduct(p);
    }
    ~scoped_logger_context() {
        Logger::SetProduct(none);
    }
};

{
    scoped_logger_context(product);
    Logger::Message(...);
}
~~~

Not quite as elegant, but I like to you use scope for these things if appropriate.
